src/NLog/NLogTraceListener.cs

```diff
@@ -251,20 +248,32 @@ public override void TraceData(TraceEventCache eventCache, string source, TraceE
             if (Filter != null && !Filter.ShouldTrace(eventCache, source, eventType, id, string.Empty, null, null, data))
                 return;
 
-            var sb = new StringBuilder();
-            for (int i = 0; i < data.Length; ++i)
+            string message = string.Empty;
+            if (data?.Length > 0)
             {
-                if (i > 0)
+                if (data.Length == 1)
                 {
-                    sb.Append(", ");
+                    message = "{0}";
+                }
+                else
+                {
+                    var sb = new StringBuilder();
```

Comment:
maybe set also the initial capacity? (5*(i+1))

Reply:
Yes good idea.